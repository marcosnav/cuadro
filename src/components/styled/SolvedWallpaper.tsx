import styled from 'styled-components';
import AppWrapper from './AppWrapper';

interface IProps {
  image?: string;
  show?: boolean;
}

const SolvedWallpaper = styled(AppWrapper)<IProps>`
  background: ${({ theme }) => theme.PURPLE};
  background-image: url("${({ image }) => image}");
  background-size: cover;
  background-position: center center;
  display: flex;
  flex-direction: column;
  left: 0;
  opacity: ${({ show }) => show ? '1' : '0'};
  top: 0;
  transition: opacity 1.2s ease;

  & > * {
    margin-bottom: 0px;
  }

  & > img {
    margin-bottom: 16px;
  }

  &::before,
  &::after {
    background: linear-gradient(180deg, rgba(0,0,0,1) 0%, rgba(0,0,0,0) 100%);
    content: '';
    height: 134px;
    left: 0;
    right: 0;
    position: absolute;
    top: 0;
    z-index: -1;
  }

  &::after {
    background: linear-gradient(0deg, rgba(0,0,0,1) 0%, rgba(0,0,0,0) 100%);
    bottom: 0;
    top: auto;
  }
`;

SolvedWallpaper.displayName = 'SolvedWallpaper';

export default SolvedWallpaper;
