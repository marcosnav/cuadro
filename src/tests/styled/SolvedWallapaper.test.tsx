import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import SolvedWallpaper from './../../components/styled/SolvedWallpaper';

describe('<SolvedWallapaper />', () => {
  test('renders properly', () => {
    const component = mount(<SolvedWallpaper image={'/cat-picture.jpg'} />);
    expect(component).toHaveStyleRule('background-image', 'url("/cat-picture.jpg")');
    expect(component).toHaveStyleRule('opacity', '0');
    expect(component).toMatchSnapshot();
  });

  test('show: true', () => {
    const component = mount(<SolvedWallpaper image={'/cat-picture.jpg'} show={true} />);
    expect(component).toHaveStyleRule('opacity', '1');
    expect(component).toMatchSnapshot();
  });
});
