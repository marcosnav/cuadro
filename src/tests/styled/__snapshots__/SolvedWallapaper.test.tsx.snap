// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SolvedWallapaper /> renders properly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: linear-gradient(180deg,undefined 0%,undefined 100%);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  left: 0;
  padding: 26px;
  position: fixed;
  top: 0;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
  width: 100%;
  background-image: url("/cat-picture.jpg");
  background-size: cover;
  background-position: center center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  left: 0;
  opacity: 0;
  top: 0;
  -webkit-transition: opacity 1.2s ease;
  transition: opacity 1.2s ease;
}

.c0 > * {
  margin-bottom: 0px;
}

.c0 > img {
  margin-bottom: 16px;
}

.c0::before,
.c0::after {
  background: linear-gradient(180deg,rgba(0,0,0,1) 0%,rgba(0,0,0,0) 100%);
  content: '';
  height: 134px;
  left: 0;
  right: 0;
  position: absolute;
  top: 0;
  z-index: -1;
}

.c0::after {
  background: linear-gradient(0deg,rgba(0,0,0,1) 0%,rgba(0,0,0,0) 100%);
  bottom: 0;
  top: auto;
}

<SolvedWallpaper
  image="/cat-picture.jpg"
>
  <div
    className="c0"
  />
</SolvedWallpaper>
`;

exports[`<SolvedWallapaper /> show: true 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: linear-gradient(180deg,undefined 0%,undefined 100%);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  left: 0;
  padding: 26px;
  position: fixed;
  top: 0;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
  width: 100%;
  background-image: url("/cat-picture.jpg");
  background-size: cover;
  background-position: center center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  left: 0;
  opacity: 1;
  top: 0;
  -webkit-transition: opacity 1.2s ease;
  transition: opacity 1.2s ease;
}

.c0 > * {
  margin-bottom: 0px;
}

.c0 > img {
  margin-bottom: 16px;
}

.c0::before,
.c0::after {
  background: linear-gradient(180deg,rgba(0,0,0,1) 0%,rgba(0,0,0,0) 100%);
  content: '';
  height: 134px;
  left: 0;
  right: 0;
  position: absolute;
  top: 0;
  z-index: -1;
}

.c0::after {
  background: linear-gradient(0deg,rgba(0,0,0,1) 0%,rgba(0,0,0,0) 100%);
  bottom: 0;
  top: auto;
}

<SolvedWallpaper
  image="/cat-picture.jpg"
  show={true}
>
  <div
    className="c0"
  />
</SolvedWallpaper>
`;
